### ISSUE TITLE
Explore the unified runtime

### ISSUE THREAD

**Issue by maleadt** [JuliaGPU/oneAPI.jl#333]
https://oneapi-src.github.io/unified-runtime/
https://github.com/oneapi-src/unified-runtime

**Comment by pengtu** [comment]
Good idea!  The memory pool and event pool from the unified runtime will be useful for resource management.
